fakestorage/upload.go

```diff
@@ -197,6 +197,11 @@ func (s *Server) multipartUpload(bucketName string, w http.ResponseWriter, r *ht
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
+	// If the backend created a Generation #, add it to the response
+	newObj, _ := s.GetObject(obj.BucketName, obj.Name)
+	obj.Generation = newObj.Generation
```

Comment:
I think it makes more sense to modify createObject to return the created object. What do you think?

Reply:
Agreed - I didn't want to introduce any larger changes though. I'll update accordingly.

Speaking of larger changes, is there a reason you didn't reuse the [Object](https://godoc.org/google.golang.org/api/storage/v1#Object) type provided by the Google API code instead of re-creating your own `Object` type? I'd be willing to tackle this conversion as well.

Reply:
I don't remember exactly why, maybe the size of the struct or something like that. Let's keep the custom Object type as changing that would be an API breaking change.

Reply:
Ok - I added the changes for having `createObject` return the created object. 

There should be virtually 100% overlap between the Object types - I think some data captured in `fake-gcs-server` is different/omitted in the JSON as compared to the original `Object` (e.g. `Bucket` vs `BucketName`). Might be worth considering, would have also caught the encoding issue with the `Generation` attribute.